src/services/crons/week.timekeeping.cache.warmer.service.ts

```diff
@@ -56,41 +56,57 @@ export class WeekTimekeepingCacheWarmerService {
                     firstWeekStartEpoch,
                 ),
             ]);
+            await this.deleteCacheKeys(abiCacheKeys);
+
+            await this.cacheWeekTimekeepingWeeklyEpochs(address, currentWeek);
+        }
+
+        profiler.stop();
+        this.logger.info(
+            `Finish refresh week timekeeping data in ${profiler.duration}`,
+            {
+                context: 'WeekTimekeepingCacheWarmer',
+            },
+        );
+    }
+
+    private async cacheWeekTimekeepingWeeklyEpochs(
+        address: string,
+        currentWeek: number,
+    ): Promise<void> {
+        const startWeek = currentWeek - constantsConfig.USER_MAX_CLAIM_WEEKS;
+
+        for (let week = startWeek; week <= currentWeek; week++) {
+            if (week < 1) {
+                continue;
+            }
 
             const [startEpochForWeek, endEpochForWeek] = await Promise.all([
                 this.weekTimekeepingCompute.computeStartEpochForWeek(
                     address,
-                    currentWeek,
+                    week,
                 ),
                 this.weekTimekeepingCompute.computeEndEpochForWeek(
                     address,
-                    currentWeek,
+                    week,
                 ),
             ]);
 
             const computeCacheKeys = await Promise.all([
                 this.weekTimekeepingSetter.startEpochForWeek(
                     address,
-                    currentWeek,
+                    week,
                     startEpochForWeek,
                 ),
                 this.weekTimekeepingSetter.endEpochForWeek(
                     address,
-                    currentWeek,
+                    week,
```

Comment:
You can calculate the `endEpochForWeek` from `startEpochForWeek`. In this way you can eliminate 1 promise and call for `startEpochForWeek` done in the end week compute method